bindings/node/native/src/tokenizer.rs

```diff
@@ -402,6 +411,20 @@ declare_types! {
                 PaddingStrategy::BatchLongest
             };
 
+            let params_object = JsObject::new(&mut cx);
+            if let Some(max_length) = max_length {
+                let obj_length = cx.number(max_length as f64);
+                params_object.set(&mut cx, "maxLength", obj_length).unwrap();
+            }
+            let obj_pad_id = cx.number(pad_id);
+            let obj_pad_type_id = cx.number(pad_type_id);
+            let obj_pad_token = cx.string(&pad_token);
+            let obj_direction = cx.string(&direction);
```

Comment:
@n1t0 I'm really puzzled here:
- I had to create the `params_object` before calling `tokenizer.with_padding`, given I was "losing" `direction` when calling it otherwise
- But... in [`setTruncation`](https://github.com/huggingface/tokenizers/pull/126/files#diff-b9e53414da9657bdf6a3b1db08f83eacR348) I was able to create the `params_object` _after_ the call to `tokenizer.with_truncation`, even though I was expecting the same error with `strategy` than with `direction`

Would you have an explanation of why I got an error in one case, and not in the other? 🤯

Reply:
Ok, so what happens here is that when you create a `PaddingParams` or a `TruncationParams`, you are giving it ownership of all the properties. So when you give it the direction or the strategy during creation, you are not supposed to be able to use them anymore. In these cases, if you want to use some variable again later, you are supposed to clone it. 
Now, there is actually an exception: when a variable is `Copy`, if you give it somewhere and keeps using it afterwards, then it will simply get copied. No need to clone it manually.
So the difference between `TruncationStrategy` and `PaddingDirection` comes from the fact that `TruncationStrategy` derives `Copy`, but not `PaddingDirection`.
https://github.com/huggingface/tokenizers/blob/d4fc17d5bc487127df299a76234b0d4f886ff200/tokenizers/src/utils.rs#L46-L51
https://github.com/huggingface/tokenizers/blob/d4fc17d5bc487127df299a76234b0d4f886ff200/tokenizers/src/tokenizer/encoding.rs#L4-L8

Reply:
Haaaaaaaaaaaaaaaaaa that's it, it's the `Copy`, thank you!!!!!
I got that it was an ownership problem, but couldn't figure why there wasn't any with `TruncationStrategy`!

Any reason why in the `TruncationStrategy` case it derives `Copy` and `PartialEq` and not in `PaddingDirection`? Should I derive `Copy` for the latter one too to make code a bit cleaner?

Reply:
OK so from https://doc.rust-lang.org/std/marker/trait.Copy.html#when-should-my-type-be-copy: 

> if your type can implement Copy, it should

I pushed a922560 with this modification, let me know if it's ok to merge for you!